exercises/complex-numbers/src/ComplexNumbers.hs

```diff
@@ -0,0 +1,41 @@
+module ComplexNumbers
+(Complex(..),
+ conjugate,
+ abs,
+ real,
+ imaginary,
+ mul,
+ add,
+ sub,
+ div) where
+
+import Prelude hiding (div, abs) -- you have to redifine this two functions
+
+-- Data definition -------------------------------------------------------------
+data Complex = Complex Float Float deriving (Eq, Show)
```

Comment:
I would not put this here, because students might think they are not allowed to change it. I would like to let the students have their creativity, to choose what they like. So, this line would probably have to be `data Complex = Dummy deriving (Eq, Show)`. Maybe even `data Complex = Dummy` if we can get away with it.

Reply:
But what will happen if someone uses a definition like ```data Complex = Complex double double``` I think that would be ok, but the tests wouldn't work.

Reply:
That's true. You could look at https://github.com/exercism/haskell/blob/master/exercises/space-age/test/Tests.hs if you want help with that, I suppose. There are example https://github.com/exercism/haskell/blob/master/exercises/space-age/examples/success-double/src/SpaceAge.hs which uses doubles and https://github.com/exercism/haskell/blob/master/exercises/space-age/examples/success-rational/src/SpaceAge.hs which uses rational - both compile with those tests. You could possibly do something similar.